app/exawind/exawind.cpp

```diff
@@ -77,30 +77,47 @@ int main(int argc, char** argv)
                       << std::endl;
     }
 
-    MPI_Comm amr_comm =
-        exawind::create_subcomm(MPI_COMM_WORLD, num_awind_ranks, 0);
-    MPI_Comm nalu_comm = exawind::create_subcomm(
-        MPI_COMM_WORLD, num_nwind_ranks, psize - num_nwind_ranks);
-
     const YAML::Node doc(YAML::LoadFile(inpfile));
     const YAML::Node node = doc["exawind"];
     const std::string amr_inp = node["amr_wind_inp"].as<std::string>();
-    const std::string nalu_inp = node["nalu_wind_inp"].as<std::string>();
     std::filesystem::path fpath_amr_inp(amr_inp);
     const std::string amr_log =
         fpath_amr_inp.replace_extension(".log").string();
     std::ofstream out(amr_log);
 
+    const auto nalu_inps = node["nalu_wind_inp"].as<std::vector<std::string>>();
+    const int num_nwsolvers = nalu_inps.size();
+    if (num_nwind_ranks % num_nwsolvers != 0) {
+        throw std::runtime_error(
+            "Number of Nalu-Wind ranks is not divisible by the number of "
+            "Nalu-Wind solver instances.");
+    }
+    const int num_nwind_ranks_per_instance = num_nwind_ranks / num_nwsolvers;
+
+    MPI_Comm amr_comm =
+        exawind::create_subcomm(MPI_COMM_WORLD, num_awind_ranks, 0);
+    std::vector<MPI_Comm> nalu_comms;
+    for (int i = 0; i < num_nwsolvers; i++) {
+        nalu_comms.push_back(exawind::create_subcomm(
+            MPI_COMM_WORLD, num_nwind_ranks_per_instance,
+            psize - num_nwind_ranks + i * num_nwind_ranks_per_instance));
```

Comment:
Does this attempt to put "I" adjacent ranks to the same sub communicator? I guess we have no way to guarantee they'll be on the same node?

Reply:
So that would be super cool. What I would love is someway to figure out the communication pattern and then rearrange the ranks of the communicators so that things happen "adjacently". As is stands, it just puts the ranks as: `[amr-wind ranks, nw first instance ranks, ..., nw nth instance ranks]`. Maybe we can do something smarter/more fun for PP22?

Reply:
That should be straightforward. I have some MPI code that can figure out which node is 
which. We can use something like this:
https://github.com/PaulMullowney/hypre-mini-app/blob/master/src/main.cpp#L9